### PR TITLE
AG-8036 Chart background image

### PR DESCRIPTION
**Pull request by alexanderby**

### CODE REVIEW

File: charts-community-modules/ag-charts-community/src/chart/background.ts [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
-    get fill(): string | undefined {
-        return this._fill;
+    @ProxyPropertyOnWrite('rectNode', 'fill')
+    fill: string | undefined;
+
+    @ActionOnSet<Background>({
+        newValue(image: BackgroundImage | undefined) {
+            if (!image) {
+                return;
+            }
```
Comment (alantreadway) [inline charts-community-modules/ag-charts-community/src/chart/background.ts:32]: You can remove these guards, the decorator only invokes these if there is a non-`undefined` value (the intent was to deliberately simplify the code for these sorts of cases)

File: charts-community-modules/ag-charts-community/src/chart/background.ts [hunk 2]
```diff
@@ -1,40 +1,53 @@
...
+            this.node.appendChild(image.node);
+            image.onload = this.imageLoadCallback;
+        },
+        oldValue(image: BackgroundImage | undefined) {
+            if (!image) {
+                return;
+            }
```
Comment (alantreadway) [inline charts-community-modules/ag-charts-community/src/chart/background.ts:39]: Ditto

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-layout/index.md [hunk 3]
```diff
@@ -76,6 +76,15 @@ By default, the chart will resize automatically to fill the container element. I
...
 
+
+## Background
+
+The chart's background can be managed via the `background.fill` and/or `background.image` properties. The `url` is required for the image. It is positioned in the center by default. `left`, `top`, `right` and `bottom` properties specify the distance between the respective borders of the image. `width` and `height` override the image size. Please see the API reference for more info.
+
+### API Reference
+
+<interface-documentation interfaceName='AgChartBackground' config='{ "showSnippets": false, "lookupRoot": "charts-api" }'></interface-documentation>
+
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-layout/index.md:87]: Could we add a new docs section, similar to Overlay's for this feature perhaps?

**Review by alantreadway — COMMENTED** [review]
Nice choice of test image :-)

File: charts-community-modules/ag-charts-community/src/chart/background.ts (hunk 1 above)
Comment (alexanderby) [reply charts-community-modules/ag-charts-community/src/chart/background.ts:32]: What if someone wants to clear the background image?

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-background/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+## Background Image
+
+The `url` is required for the background image.
+
+It is positioned in the center by default. The `left`, `top`, `right` and `bottom` properties specify the distance between the chart borders and the image. The `width` and `height` properties override the size of the image.
+
+```js
+background: {
+    image: {
+        url: 'data:image/png;base64,iVBORw0...',
```
Comment (alantreadway) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-background/index.md:28]: I think we may need to at least elaborate on this possibly being a `http://` URL and any behaviours related to doing that, but let's get this change merged for feedback from David

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**